benchmark.py

```diff
@@ -546,7 +632,7 @@ def main(phase, host, port, user, password, database, data_dir, query_root, dbge
     parser.add_argument("-q", "--query-root", default = "./query_root", help = "Directory for query files; default is ./query_root")
     parser.add_argument("-g", "--dbgen-dir", default = "./tpch-dbgen", help = "Directory containing tpch dbgen source; default is ./tpch-dbgen")
     parser.add_argument("-s", "--scale", type = float, default = 1.0, help = "Size of the data generated; default is 1.0 = 1GB")
-    parser.add_argument("-n", "--num-streams", type = float, default = 1, help = "Number of streams to run the throughput test with; default is 1")
+    parser.add_argument("-n", "--num-streams", type = int, default = 1, help = "Number of streams to run the throughput test with; default is 1")
```

Comment:
Of course. It is expected that it should be the same. If you want to run with 3 streams you should generate data for at least 3 streams. However, you can still generate data for 4 streams and run only with 2. I am not sure what a good solution would be hence I left the control to the user.

Reply:
Hm. Looking at the page 6 Table 2 in [iceis2012.pdf](https://github.com/Data-Science-Platform/tpch-pgsql/blob/master/iceis2012.pdf) I would say, that number of query streams (S) is dependent on scale factor (SF). So if we use SF 1, we should set S to 2. But because we want to run it twice, then we should actually run QGen with 4. But user can still overwrite this value.

Reply:
I don't think query streams are dependent on the scale factor. From my understanding, the only thing dependent on the scale factor is the update size for the refresh streams which is 10% (`dbgen` allows to change this but we have it hardcoded for simplicity) of the scale factor size. But this is for every refresh stream. So if you have 10 streams each would insert/delete data equaling 10% of the scale factor. The number of streams itself is not related to the scale. The table that you list just points out what those guys ran. I don't think this is part of the specification.